File: TDDMicroExercises.ProposedSolution/Javascript/tire-pressure-monitoring-system/alarm.js

Alarm = function(sensor) {

	this._lowPressureTreshold = 17;
	this._highPressureTreshold = 21;
	this._sensor = sensor;
	this._alarmOn = false;
	this._alarmCount = 0;
};

Alarm.Create = function() {
	return new Alarm(new Sensor());
};

Alarm.prototype = {

	check: function () {
		
		var psiPressureValue = this._sensor.popNextPressurePsiValue();

		if (psiPressureValue < this._lowPressureTreshold || this._highPressureTreshold < psiPressureValue)
		{
			this._alarmOn = true;
			this._alarmCount += 1;
		}
	},
	
	alarmOn: function () {
		 return this._alarmOn;
	}

};

File: Swift/Sources/TirePressure/Alarm.swift
import Foundation

public class Alarm {
    private let lowPressureThreshold: Double = 17
    private let highPressureThreshold: Double = 21

    let _sensor = Sensor()

    private var _alarmOn = false
    private var _alarmCount = 0

    public func check() {
        let psiPressureValue = _sensor.PopNextPressurePsiValue()

        if (psiPressureValue > lowPressureThreshold || highPressureThreshold < psiPressureValue) {
            _alarmOn = true
            _alarmCount += 1
        }
    }

    public var alarmOn: Bool { _alarmOn }
}
